src/labeler.ts

```diff
@@ -213,6 +223,42 @@ function checkAll(changedFiles: string[], globs: string[]): boolean {
   return true;
 }
 
+function matchBranchPattern(matcher: IMinimatch, branchName: string): boolean {
+  core.debug(`  - ${printPattern(matcher)}`);
+  if (matcher.match(branchName)) {
+    core.debug(`   "branch" pattern matched`);
+    return true;
+  }
+
+  core.debug(`   ${printPattern(matcher)} did not match`);
+  return false;
+}
+
+function checkBranch(glob: string | string[]): boolean {
+  const branchName = getBranchName();
+  if (!branchName) {
+    core.debug(` no branch name`);
+    return false;
+  }
+
+  core.debug(` checking "branch" pattern against ${branchName}`);
+  if (Array.isArray(glob)) {
+    const matchers = glob.map(g => new Minimatch(g));
```

Comment:
You are using the minimatch library under the hood, and it works fine, but as far as I know, it is designed mainly to work with paths, and its syntax may not be obvious to some users for branch filtering. I think the best option here would be to just use regular expressions. We have a [feature request](https://github.com/actions/labeler/issues/55) to add labels based on the title and description of the pull request and we could prepare a foundation for this functionality in scope of your PR.